apps/glusterfs/app_topology_test.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/heketi/tests"
+)
+
+func TestTopologyInfo(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	// Create the app
+	app := NewTestApp(tmpfile)
+	defer app.Close()
+
+	err := setupSampleDbWithTopology(app,
+		1,      // clusters
+		2,      // nodes_per_cluster
+		1,      // devices_per_node,
+		500*GB, // disksize)
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a volume which uses the entire storage
+	v := createSampleReplicaVolumeEntry(495, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = v.Create(app.db, app.executor, app.Allocator())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	topologyInfo, err := app.TopologyInfo()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testCluster := topologyInfo.ClusterList[0]
+
+	tests.Assert(t, testCluster.Id != "")
+	tests.Assert(t, len(testCluster.Volumes) > 0)
+	tests.Assert(t, len(testCluster.Nodes) > 0)
```

Comment:
Somewhat minor nit: It's much nicer if we also provide an "expected string". Something like:
```golang
tests.Assert(t, testCluster.Id != "", `expected testCluster.Id != "", got`, testCluster.Id)
```
It is annoying to type but much better if you come back 6 weeks later or have to fix a failing test somone else wrote.